src/graph_gui.cpp

```diff
@@ -109,19 +109,19 @@ struct GraphLegendWindow : Window {
  * Construct a vertical list of buttons, one for each company.
  * @return Panel with company buttons.
  */
-static NWidgetBase *MakeNWidgetCompanyLines()
+static std::unique_ptr<NWidgetBase> MakeNWidgetCompanyLines()
 {
-	NWidgetVertical *vert = new NWidgetVertical(NC_EQUALSIZE);
+	std::unique_ptr<NWidgetVertical> vert = std::make_unique<NWidgetVertical>(NC_EQUALSIZE);
```

Comment:
This is one of those places where I'd use auto, rather than just repeating the type